disk/disk.go

```diff
@@ -34,7 +39,7 @@ func WriteAt(path string, p []byte, off int64) (int, error) {
 		data = append(make([]byte, padding), p...)
 	}
 	// index that ends writing
-	end := st + int64(len(data))
+	end := st + int64(len(data)) - 1
```

Comment:
Let's say I to write one block(`payloadSize == 4096 - crcSize`) into a empty file(`st == 0`), `endIdx == 1` by calculate `(st + int64(len(data)))/psize` but it should be `0`